pkg/graveler/ref/manager.go

```diff
@@ -71,9 +72,27 @@ func NewKVRefManager(executor batch.Batcher, kvStore kv.StoreMessage, addressPro
 	}
 }
 
+func (m *KVManager) updateRepoState(ctx context.Context, repo *graveler.RepositoryRecord, state graveler.RepositoryState) error {
```

Comment:
I think some of these usages should be `SetIf`

Reply:
For now, I don't see a scenario for SetIf, it will also require us to have another Get or pass the predicate to this function.
I suggest handling this when we have a valid scenario for SetIf